zarr/util.py

```diff
@@ -365,3 +367,142 @@ def __repr__(self):
     def _repr_html_(self):
         items = self.obj.info_items()
         return info_html_report(items)
+
+
+class ZarrGroupTraversal(Traversal):
+
+    def get_children(self, node):
+        return getattr(node, "values", lambda: [])()
+
+    def get_root(self, tree):
+        return tree
+
+    def get_text(self, node):
+        name = node.name.split("/")[-1] or "/"
+        name += "[...]" if hasattr(node, "dtype") else ""
+        return name
+
+
+def custom_html_sublist(group, indent):
+    traverser = ZarrGroupTraversal(tree=group)
+    result = ""
+
+    result += (
+        """{0}<li><div>{1}</div>""".format(
+            indent, traverser.get_text(group)
+        )
+    )
+
+    children = traverser.get_children(group)
+    if children:
+        result += """\n{0}{0}<ul>\n""".format(indent)
+    for c in children:
+        for l in custom_html_sublist(c, indent).splitlines():
+            result += "{0}{0}{1}\n".format(indent, l)
+    if children:
+        result += "{0}{0}</ul>\n{0}".format(indent)
+
+    result += (
+        """</li>\n""".format(
+            indent, traverser.get_text(group)
+        )
+    )
+
+    return result
+
+
+def custom_html_list(group, indent="    "):
+    result = ""
+
+    # Add custom CSS style for our HTML list
+    result += """<style type="text/css">\n"""
+    result += dedent("""\
+        div.zarrTree ul,
+        div.zarrTree li,
+        div.zarrTree li > div {
+            display: block;
+            position: relative;
+        }
+
+        div.zarrTree ul,
+        div.zarrTree li {
+            list-style-type: none;
+        }
+
+        div.zarrTree li {
+            border-left: 2px solid #000;
+            margin-left: 1em;
+        }
+
+        div.zarrTree li > div {
+            padding-left: 1.3em;
+            padding-top: 0.225em;
+            padding-bottom: 0.225em;
+        }
+
+        div.zarrTree li > div::before {
+            content: '';
+            position: absolute;
+            top: 0;
+            left: -2px;
+            bottom: 50%;
+            width: 1.2em;
+            border-left: 2px solid #000;
+            border-bottom: 2px solid #000;
+        }
+
+        div.zarrTree > ul > li:first-child > div {
+            padding-left: 4%;
+        }
+
+        div.zarrTree > ul > li:first-child > div::before {
+            border: 0 none transparent;
+        }
+
+        div.zarrTree ul > li:last-child {
+            border-left: 2px solid transparent;
+        }
+    """)
+    result += "</style>\n\n"
+
+    # Insert the HTML list
+    result += """<div class="zarrTree">\n"""
```

Comment:
Nit, could you make the CSS class "zarr-tree" for consistency of naming with "zarr-info".